packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -50,6 +50,13 @@
 
 #include <Kokkos_ArithTraits.hpp>
 #include <KokkosBatched_Util.hpp>
+
+/// complex is_convertible override
+namespace std {
+  template<typename T1, typename T2> struct is_convertible<Kokkos::complex<T1>, T2> { static constexpr bool value = std::is_convertible<T1,T2>::value; };
```

Comment:
It's forbidden to override `std::is_convertible`: https://github.com/kokkos/kokkos/issues/1658#issuecomment-394508562

It should also be unnecessary.

Reply:
Then, this needs to wait until the next kokkos kernels promotion. I put the static_assert in kokkoskernels in SIMD constructor. @lucbv

Reply:
@kyungjoo-kim submitted PR kokkos/kokkos-kernels#251 as a fix for the `is_convertible` troubles; following spot-check, a matching patch to Trilinos can be issued as a hotfix. 
For bookkeeping may be nicer to issue the PR separately from this one, but that increases autotester time so may be more efficient to just include it here.